src/gulp/tasks/ts-lint.js

```diff
@@ -17,6 +18,7 @@ class TsLintTaskLoader extends AbstractTaskLoader {
 
         gulp.task("ts-lint", "Lint TypeScript code", () =>{
             let src = null;
+            const noNgFactory = filter([ "**/*.ts", "!**/*.ngfactory.ts" ]);
```

Comment:
Better to add the ngfactory file type to the config constants. Also use the typescript glob from config as well e.g. config.globs.scripts.typescript, can't remember if this has **/ by default, but there is a setting for it I'm sure.